common/src/main/java/com/microsoft/identity/common/components/AndroidPlatformComponentsFactory.java

```diff
@@ -0,0 +1,123 @@
+import com.microsoft.identity.common.internal.providers.oauth2.AndroidTaskStateGenerator;
+import com.microsoft.identity.common.internal.ui.AndroidAuthorizationStrategyFactory;
+import com.microsoft.identity.common.java.WarningType;
+import com.microsoft.identity.common.java.interfaces.IPlatformComponents;
+import com.microsoft.identity.common.java.interfaces.PlatformComponents;
+import com.microsoft.identity.common.java.net.DefaultHttpClientWrapper;
+import com.microsoft.identity.common.java.platform.Device;
+import com.microsoft.identity.common.logging.Logger;
+
+import lombok.NonNull;
+
+/**
+ * A factory class for building Android implementations of platform-dependent components in Common.
+ */
+public class AndroidPlatformComponentsFactory {
+    /**
+     * True if all of the platform-dependent static classes have been initialized.
+     */
+    private static boolean sGlobalStateInitalized = false;
+
+    /**
+     * Initializes platform-dependent static classes.
+     */
+    public static synchronized void initializeGlobalStates(@NonNull final Context context){
+        if (!sGlobalStateInitalized) {
+            HttpCache.initialize(context);
+            Device.setDeviceMetadata(new AndroidDeviceMetadata());
+            Logger.setAndroidLogger();
+            sGlobalStateInitalized = true;
+        }
+    }
+
+    /**
+     * Creates an {@link IPlatformComponents} object from a {@link Context}.
+     *
+     * @param context an application context.
+     **/
+    public static IPlatformComponents createFromContext(@NonNull final Context context) {
+        return create(context, null, null);
+    }
+
+    /**
+     * Creates an {@link IPlatformComponents} object from an {@link Activity} and, optionally, a {@link Fragment}.
+     *
+     * @param activity an activity where an interactive session will be attached to.
+     * @param fragment a fragment where an interactive session will be attached to.
+     **/
+    public static IPlatformComponents createFromActivity(@NonNull final Activity activity,
+                                                               @Nullable final Fragment fragment) {
+
+        return create(activity.getApplicationContext(), activity, fragment);
+    }
+
+    @SuppressWarnings(WarningType.rawtype_warning)
+    private static IPlatformComponents create(@NonNull final Context context,
+                                              @Nullable final Activity activity,
+                                              @Nullable final Fragment fragment) {
+        initializeGlobalStates(context);
+
+        final PlatformComponents.PlatformComponentsBuilder builder = PlatformComponents.builder();
+        fillBuilder(builder, context, activity, fragment);
+        return builder.build();
+    }
+
+    @SuppressWarnings(WarningType.rawtype_warning)
+    public static void fillBuilder(@NonNull final PlatformComponents.PlatformComponentsBuilder builder,
```

Comment:
I don't really understand the purpose of this method. Can you add some javadoc?
Also is this meant to be `public`?

Reply:
I made AndroidBrokerPlatformComponentFactory invoke this method when it constructs its own object.

Didn't make them inherit each other since they're static factory classes. Maybe I can un-static them.

Reply:
javadoc added.